pixman/src/region.rs

```diff
@@ -204,14 +215,14 @@ macro_rules! impl_region {
             }
 
             /// Whether this region contains the provided rectangle
-            pub fn contains_rectangle(&self, rect: $box_t) -> Option<usize> {
+            pub fn contains_rectangle(&self, rect: $box_t) -> Option<Overlap> {
                 let overlap =
                     unsafe { paste!($crate::ffi::[<$impl _contains_rectangle>](&self.0, &rect)) };
-
-                if overlap > 0 {
-                    Some(overlap as usize)
-                } else {
-                    None
+                match overlap {
+                    crate::ffi::pixman_region_overlap_t_PIXMAN_REGION_OUT => Some(Overlap::Out),
+                    crate::ffi::pixman_region_overlap_t_PIXMAN_REGION_IN => Some(Overlap::In),
+                    crate::ffi::pixman_region_overlap_t_PIXMAN_REGION_PART => Some(Overlap::Part),
```

Comment:
This should use `$crate::ffi::`

Looking at the implementation in pixman I think it would also be fine to remove the `Option` and use `unreachable!()` as the general catcher.